pkg/runner/runner.go

```diff
@@ -94,13 +97,28 @@ func ParseFlags(cfg *Config) (*Config, error) {
 	} else {
 		return nil, fmt.Errorf("Invalid option for migrate: %v. Valid options are [true, false, only]", migrateOption)
 	}
+
+	if cfg.ReadOnly {
+		if cfg.HaGroupLockID != 0 {
+			return nil, fmt.Errorf("Invalid option for HA group lock ID, cannot enable HA mode and read-only mode")
+		}
+		cfg.Migrate = false
+		cfg.StopAfterMigrate = false
+		cfg.UseVersionLease = false
+		cfg.InstallTimescaleDB = false
+		cfg.PgmodelCfg.ReadOnly = true
```

Comment:
If the user provides input for these, we should I guess error then and say that these are not supported, else users will be confused of the flag as they provided but we disabled it. A generic error like `Flag not supported in read-only`

Reply:
Added checks.